cloudinit/distros/alpine.py

```diff
@@ -162,4 +162,31 @@ def preferred_ntp_clients(self):
 
         return self._preferred_ntp_clients
 
+    def shutdown_command(self, mode='poweroff', delay='now', message=None):
+        # called from cc_power_state_change.load_power_state
+        # Alpine has halt/poweroff/reboot, with the following specifics:
+        # - we use them rather than the generic "shutdown"
+        # - delay is given with "-d [integer]"
+        # - the integer is in seconds, cannot be "now", and takes no "+"
+        # - no message is supported (argument ignored, here)
+
+        command = [mode, "-d"]
+
+        # Convert delay from minutes to seconds, as Alpine's
+        # halt/poweroff/reboot commands take seconds rather than minutes.
+        if delay == "now":
+            # Alpine's commands do not understand "now".
+            command += ['0']
+        else:
+            # No "+" in front of delay value (not supported)
+            try:
+                command += ['%d' % int(int(delay) * 60)]
+            except ValueError as e:
+                raise TypeError(
+                    "power_state[delay] must be 'now' or '+m' (minutes)."
```

Comment:
I'm leaving it as it is. Yes I agree that the code is liberal in what requires. But having the documentation as well as the possibly reported error messages consistent with what might be written elsewhere is also good.

Reply:
> (You didn't introduce this wording (and I think it _was_ strictly true until #535 landed with Alpine support), so I'm not asking you to change it, just flagging it up as something I noticed.)

Hi. That wording was unchanged by my #535 submission. Checking the Debian manpage for shutdown it states:

```
The time string may either be in the format "hh:mm" for hour/minutes specifying the
time to execute the shutdown at, specified in 24h clock format. Alternatively it may
be in the syntax "+m" referring to the specified number of minutes m from now.
```
which fits in with that error message. I did tweak the documentation section at the top of the module to remove the reference to "the most common format is", should I have left it as it was?

Reply:
The (minor) discrepancy was introduced because https://github.com/canonical/cloud-init/commit/79a8ce7e714ae1686c10bff77612eab0f6eccc95#diff-b932b4a5cf18b1d52fbdca347693cd86L170 dropped the enforcement of the leading `+` via regex:

```py
if delay != "now" and not re.match(r"\+[0-9]+", delay):
```

to just passing it into `int` after . This makes us more liberal in what we accept (which IMO is a good thing), it just means that "must be ... '+m'" is now more strict than the code itself actually is.

I don't think it's a particularly big deal, and looking at the code again I think the message was actually already too strict before #535: we passed the user-specified option through `int` in the old code path too, before prepending "+".